clusterctl/examples/google/provider-components.yaml.template

```diff
@@ -0,0 +1,354 @@
+          Environment="KUBELET_EXTRA_ARGS=--cloud-provider=gce --cloud-config=/etc/kubernetes/cloud-config"
+          EOF
+          systemctl daemon-reload
+          systemctl restart kubelet.service
+          PRIVATEIP=`curl_metadata "network-interfaces/0/ip"`
+          echo $PRIVATEIP > /tmp/.ip
+          PUBLICIP=`curl_metadata "network-interfaces/0/access-configs/0/external-ip"`
+
+          # Set up the GCE cloud config, which gets picked up by kubeadm init since cloudProvider is set to GCE.
+          cat > /etc/kubernetes/cloud-config <<EOF
+          [global]
+          project-id = ${PROJECT}
+          network-name = ${NETWORK}
+          subnetwork-name = ${SUBNETWORK}
+          node-tags = ${NODE_TAG}
+          EOF
+
+          # Set up kubeadm config file to pass parameters to kubeadm init.
+          cat > /etc/kubernetes/kubeadm_config.yaml <<EOF
+          apiVersion: kubeadm.k8s.io/v1alpha1
+          kind: MasterConfiguration
+          api:
+            advertiseAddress: ${PUBLICIP}
+            bindPort: ${PORT}
+          networking:
+            serviceSubnet: ${SERVICE_CIDR}
+          kubernetesVersion: v${CONTROL_PLANE_VERSION}
+          cloudProvider: gce
+          token: ${TOKEN}
+          controllerManagerExtraArgs:
+            cluster-cidr: ${POD_CIDR}
+            service-cluster-ip-range: ${SERVICE_CIDR}
+            allocate-node-cidrs: "true"
+          apiServerCertSANs:
+          - ${PUBLICIP}
+          - ${PRIVATEIP}
+          EOF
+
+          function install_certificates () {
+              if ! curl_metadata "attributes/ca-cert"; then
+                  return
+              fi
+              echo "Configuring custom certificate authority..."
+              PKI_PATH=/etc/kubernetes/pki
+              mkdir -p ${PKI_PATH}
+              CA_CERT_PATH=${PKI_PATH}/ca.crt
+              curl_metadata "attributes/ca-cert" | base64 -d > ${CA_CERT_PATH}
+              chmod 0644 ${CA_CERT_PATH}
+              CA_KEY_PATH=${PKI_PATH}/ca.key
+              curl_metadata "attributes/ca-key" | base64 -d > ${CA_KEY_PATH}
+              chmod 0600 ${CA_KEY_PATH}
+          }
+
+          # Create and set bridge-nf-call-iptables to 1 to pass the kubeadm preflight check.
+          # Workaround was found here:
+          # http://zeeshanali.com/sysadmin/fixed-sysctl-cannot-stat-procsysnetbridgebridge-nf-call-iptables/
+          modprobe br_netfilter
+
+          install_certificates
+
+          kubeadm init --config /etc/kubernetes/kubeadm_config.yaml
+
+          for tries in $(seq 1 60); do
+              kubectl --kubeconfig /etc/kubernetes/kubelet.conf annotate --overwrite node $(hostname) machine=${MACHINE} && break
+              sleep 1
+          done
+          echo done.
+          ) 2>&1 | tee /var/log/startup.log
+    - machineParams:
+      - os: ubuntu-1604-lts
+        roles:
+        - Node
+        versions:
+          kubelet: 1.9.4
+          containerRuntime:
```

Comment:
no longer needed.

Reply:
Hmmm..... I am suprised this works..... Will need to do some sanity checks.....

Reply:
Removed (and it works end-to-end). I strongly suspect there may be some bug FYI - @kcoronado .